src/material/form-field/_form-field-focus-overlay.scss

```diff
@@ -19,13 +19,6 @@
       tokens-mat-form-field.$prefix, tokens-mat-form-field.get-token-slots()) {
       @include token-utils.create-token-slot(background-color, state-layer-color);
 
-      .mat-mdc-text-field-wrapper:hover & {
```

Comment:
We need to keep the hover/focus states in order to be in line with the Material spec. See https://m3.material.io/components/text-fields/specs#dbc210fc-c916-4d28-8541-f3ecdb954a62. Instead we should change the opacity to improve the contrast.

Reply:
Reinserted :hover and .mat-focused states and updated the opacity value of focus-state-layer-opacity to improve contrast as recommended.